Library/ViewModels/WatchProjectViewModel.swift

```diff
@@ -0,0 +1,267 @@
+          .map { _ in (project, project.personalization.isStarred ?? false, success: true) }
+          .flatMapError { _ in .init(value: (project, !shouldWatch, success: false)) }
+    }
+
+    let projectOnSaveButtonToggleSuccess = watchProjectResult
+      .filter(third)
+      .map(first)
+
+    // update the cache with the result and return it
+    let projectResultUpdatingCache = watchProjectResult
+      .map { project, shouldWatch, _ in (project, shouldWatch) }
+      .map(watchAndCacheProject(_:shouldWatch:))
+
+    let projectSavedFromNotification = configuredProject
+      .takePairWhen(self.projectFromNotificationProperty.signal.skipNil())
+      .filter { $0.id == $1.id }
+      .map { $0.1 }
+
+    // Project emits with initial configured value, immediately when watched, again on watch result
+    let project = Signal.merge(
+      configuredProject,
+      projectSavedFromNotification,
+      watchProjectToggle.map(first),
+      projectResultUpdatingCache.map(first)
+    )
+
+    self.goToLoginTout = loggedOutUserTappedSaveButton.ignoreValues()
+
+    self.showProjectSavedAlert = project
+      .takeWhen(saveButtonTapped)
+      .filter { !$0.endsIn48Hours(today: AppEnvironment.current.dateType.init().date ) }
+      .filter { _ in
+        !AppEnvironment.current.ubiquitousStore.hasSeenSaveProjectAlert ||
+          !AppEnvironment.current.userDefaults.hasSeenSaveProjectAlert
+      }
+      .on(value: { _ in
+        AppEnvironment.current.ubiquitousStore.hasSeenSaveProjectAlert = true
+        AppEnvironment.current.userDefaults.hasSeenSaveProjectAlert = true
+      })
+      .ignoreValues()
+
+    self.showNotificationDialog = project
+      .takeWhen(saveButtonTapped)
+      .filter { _ in shouldShowNotificationDialog() }
+      .on(value: { _ in
+        AppEnvironment.current.ubiquitousStore.hasSeenSaveProjectAlert = true
+        AppEnvironment.current.userDefaults.hasSeenSaveProjectAlert = true
+      })
+      .ignoreValues()
+      .map { _ in
+        Notification(
+          name: .ksr_showNotificationsDialog,
+          userInfo: [UserInfoKeys.context: PushNotificationDialog.Context.save]
+        )
+    }
+
+    self.saveButtonSelected = project
+      .map { $0.personalization.isStarred == true }
+      .skipRepeats()
+
+    self.generateSuccessFeedback = saveButtonTapped.signal.filter(isFalse).ignoreValues()
+    self.generateSelectionFeedback = saveButtonTapped.signal.filter(isTrue).ignoreValues()
+
+    self.saveButtonAccessibilityValue = self.saveButtonSelected
+      .map { starred in starred ? Strings.Saved() : Strings.Unsaved() }
+
+    self.postNotificationWithProject = project
+      .takeWhen(saveButtonTapped)
+
+    projectOnSaveButtonToggleSuccess
+      .observeValues { AppEnvironment.current.koala.trackProjectSave($0, context: .project) }
+  }
+
+  private let awakeFromNibProperty = MutableProperty(())
+  public func awakeFromNib() {
```

Comment:
We have `awakeFromNib()` as well as `viewDidLoad()` given that this view model is sometimes used in a cell and sometimes in a view controller.